painless_sqlalchemy/util/LocationUtil.py

```diff
@@ -0,0 +1,111 @@
+from math import radians, cos, sin, asin, sqrt, isnan, isinf
+
+
+def _validate_coordinate(coordinate, valid_range):
+    """
+        Check if a coordinate is valid. Throws error if not.
+    :param coordinate: the input
+    :param valid_range: the valid range (abs)
+    :return: the coordinate
+    """
+    try:
+        coordinate = float(coordinate)
+    except (TypeError, ValueError) as err:
+        raise ValueError('%s not a float' % str(coordinate), err)
+    if isnan(coordinate) or isinf(coordinate):
+        raise ValueError('%s can not be nan or inf' % str(coordinate))
+    if abs(coordinate) > valid_range:
+        raise ValueError(
+            '%s is not in range %s' % (str(coordinate), str(valid_range)))
+    return coordinate
+
+
+def validate_latitude(latitude):
+    """
+        Check if latitude is valid.
+    :param latitude: Latitude to check
+    :return: the latitude if valid, otherwise throws error
+    """
+    try:
+        return _validate_coordinate(latitude, 90)
+    except Exception as e:
+        raise ValueError("Invalid latitude given.", e)
+
+
+def validate_longitude(longitude):
+    """
+        Check if longitude is valid.
+    :param longitude: Longitude to check
+    :return: the longitude if valid, otherwise throws error
+    """
+    try:
+        return _validate_coordinate(longitude, 180)
+    except Exception as e:
+        raise ValueError("Invalid longitude given.", e)
+
+
+def validate_latlong(latlong):
+    """
+        Check if latlong location is valid.
+
+        Throws error if not
+    :param latlong: Location to check
+    :return: tuple (lat, long)
+    """
+    if not isinstance(latlong, (tuple, list)) or len(latlong) != 2:
+        raise ValueError("Invalid latlong format.")
+    return validate_latitude(latlong[0]), validate_longitude(latlong[1])
+
+
+def haversine(lat1, lon1, lat2, lon2):
+    """
+    Calculate the great circle distance between two points
+    on the earth (specified in decimal degrees).
+    Returns distance in km.
```

Comment:
Should we modify this to return in `m` instead? From previous in person discussion about keeping `m` as distance standard

Reply:
done